Raven.Client.Lightweight/Listeners/IDocumentSaveChangesListener.cs

```diff
@@ -0,0 +1,20 @@
+//-----------------------------------------------------------------------
+// <copyright file="IDocumentStoreListener.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+using Raven.Json.Linq;
+
+namespace Raven.Client.Listeners
+{
+    /// <summary>
+    /// Hook for users to provide additional logic on store operations
+    /// </summary>
+    /// 
+    public interface IDocumentSaveChangesListener
+    {
+        void BeforeSaveChanges(IDocumentSession session);
+
+        void AfterSaveChanges(IDocumentSession session);
+    }
```

Comment:
Consider the BeforeSaveChanges method.
How do you get the list of documents that will be modified so you can create your Commit instace?